File: api/src/main/java/com/rxdroid/api/github/provider/GitHubUserProvider.java
package com.rxdroid.api.github.provider;

import android.support.annotation.NonNull;
import com.rxdroid.api.github.GitHubApi;
import com.rxdroid.api.github.model.GitHubUserModel;
import dagger.Reusable;
import io.reactivex.Observable;
import javax.inject.Inject;
import retrofit2.Response;

@Reusable
public class GitHubUserProvider extends GitHubApiProvider implements ApiProvider<GitHubUserModel> {

    @Inject
    public GitHubUserProvider(final GitHubApi gitHubApi) {
        super(gitHubApi);
    }

    @Override
    public Observable<Response<GitHubUserModel>> loadBySearchValue(@NonNull final String searchValue) {
        return getGitHubApi().getUserByName(searchValue);
    }
}

File: api/src/main/java/com/rxdroid/api/github/provider/GitHubUserProvider.kt
package com.rxdroid.api.github.provider

import com.rxdroid.api.github.GitHubApi
import com.rxdroid.api.github.model.GitHubUserData
import dagger.Reusable
import io.reactivex.Observable
import retrofit2.Response
import javax.inject.Inject

@Reusable
class GitHubUserProvider @Inject
constructor(gitHubApi: GitHubApi) : GitHubApiProvider(gitHubApi), ApiProvider<GitHubUserData> {

    override fun loadBySearchValue(searchValue: String): Observable<Response<GitHubUserData>> {
        return gitHubApi.getUserByName(searchValue)
    }

}
